src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java

```diff
@@ -212,6 +195,20 @@ public final void setTabWidth(int tabWidth) {
         this.tabWidth = tabWidth;
     }
 
+    @Override
+    public final void log(int line, String key, Object... args) {
```

Comment:
I was able to achieve 3.0 penalty by using the following order:
1. `log(int,String,Object...)`
2. `log(int,int,String,Object...)`
3. `log(DetailAST,String,Object...)`
4. `getLines()`

Could you please check it

Reply:
Done. Thanks.

Reply:
I've updated the report :)